core/includes/install.inc

```diff
@@ -741,7 +744,10 @@ make this configuration active, see admin/config/development/configuration/sync.
 ');
     }
 
-    file_put_contents($config_directory . '/README.md', $text);
+    // Only write the README.md to a file storage directory.
+    if (is_a($config_storage, 'ConfigFileStorage')) {
+      file_put_contents(config_get_config_directory($config_type) . '/README.md', $text);
+    }
```

Comment:
Seems like this code for the READMEs could go in `ConfigFileStorage::initializeStorage()`? We'd need to pass in the config storage type (active/staging), but that seems like it would be appropriate.